contracts/evoting/types/ballots_test.go

```diff
@@ -8,17 +8,23 @@ import (
 	"github.com/stretchr/testify/require"
 )
 
-var ballot1 = string("select:" + encodedQuestionID(1) + ":1,0,1\n" +
-	"rank:" + encodedQuestionID(2) + ":1,2,0,,\n" +
-	"select:" + encodedQuestionID(3) + ":1,0,1,1\n" +
-	"text:" + encodedQuestionID(4) + ":YmxhYmxhYmxhZg==,Y2VzdG1vaUVtaQ==\n\n")
-
-var ballot2 = string("select:" + encodedQuestionID(1) + ":0,0,0\n" +
-	"rank:" + encodedQuestionID(2) + ":128,128,128,128\n" +
-	"select:" + encodedQuestionID(3) + ":0,0,0,0,0\n" +
-	"text:" + encodedQuestionID(4) + ":xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx," +
+const SELECT_STR = "select:"
+const RANK_STR = "rank:"
+const TEXT_STR = "text:"
+const UNMARSHALING_RANK_STR = "could not unmarshal rank answers: "
+const UNMARSHALING_TEXT_STR = "could not unmarshal text answers: "
```

Comment:
We use camel case for constants as well. You can also declare them altogether since they're related, see PR #183 or `contract/evoting.go` for examples

Reply:
yes, I think it would make sense and we will be able to use them in some functions in ballots.go such as unmarshal(...) or MaxEncodedSize(...). The only problem I can see is that the format is slightly different between the strings in the tests and the ones in ballots.go and that they are mostly recurrent in the tests. 
Thanks for the note on the camel case convention, I have to get rid of my Java habits :)

Reply:
Do you mean because of the `:` ? Since this is just formatting it does not have to be part of the string. It could be written as a join in the tests.
This might be overthinking it (although quite clean), but we could use the Question interface to force the Text/Rank/Select questions to re-define the "String()" function which would simply return "select"/"rank"/"text" and could be called everywhere. It's just an idea though ;)